dds/DCPS/XTypes/TypeObject.cpp

```diff
@@ -725,6 +757,19 @@ void compute_dependencies(const TypeMap& type_map,
   }
 }
 
+bool write_empty_xcdr2_nonfinal(DCPS::Serializer& strm)
+{
+  return strm.write_delimiter(0);
```

Comment:
`write_delimiter()` function is a bit confusing; I think its argument is the total size that includes 4 bytes of the Dheader itself. So the return statement would be `return strm.write_delimiter(4);` I think.

Reply:
Ah yes, I forgot about that. Almost all uses of it would need the subtraction it's doing, but yea it's a bit confusing if you're doing something manually instead of using `serialized_size`.